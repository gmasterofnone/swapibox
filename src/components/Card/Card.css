.card {
  width: 350px;
  height: 530px;
  background-color: blue;
  border: 10px solid white;
  margin: 25px;
}

.image-container {
  display: flex;
  justify-content: flex-end;
}

.character-pic {
  background-color: purple;
  height: 330px;
  width: 350px;
  z-index: 998;
  position: relative;
}

.favorite-btn {
  background-color: transparent;
  position: absolute;
  height: 50px;
  width: 50px;
  z-index: 999;
}

.star {
  margin: 10px 15px 0 0;
}

.card-info {
  align-items: center;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 10px;
}

.character-name {
  text-align: center;
  margin: 5px 0 0 0;
  padding: none;
}

ul {
  list-style: none;
  /* margin: 0; */
  padding: 0;
}
li, h4 {
  color: white;
}

