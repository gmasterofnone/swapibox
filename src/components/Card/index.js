import React, { Component } from 'react';
import './Card.css'
import cardImage from './../Utils/cardImage.js';
import { filterCards } from '../Utils/API';
import PropTypes from 'prop-types';


class Card extends Component {
  constructor() {
    super()
    this.state = {
      favorite: false
    }
  }

  componentDidMount() {
    const { favorite } = this.props.cardData
    return favorite ? this.setState({ favorite }) : null
  }

  updateFavorites = (name, getFavorites) => {
    filterCards(name)
    this.setState( {favorite: !this.state.favorite} )
    getFavorites()
  }

  render() {
    const { favorite } = this.state;
    const { cardData, getFavorites } = this.props
    console.log(typeof getFavorites)
    const { name } = this.props.cardData;
    const description = Object.keys(cardData).map((spec, index) => {
      if (spec !== 'type' && spec !== 'name' && spec !== 'favorite' && !Array.isArray(cardData[spec])) {
        return <li key={Date.now() + index}>{spec}: {cardData[spec]}</li>
      } else if (Array.isArray(cardData[spec])) {
        return <li key={Date.now() + index}>{spec}: {cardData[spec].join(', ')}</li>
      }
      return '';
    })

    return(
      <div className='card'>
        <div className='image-container'>
          <div onClick={() => this.updateFavorites(name, getFavorites)}
          className='favorite-btn'
          >
            <img className='star' alt='star' src={`${cardImage[favorite]}`}/>
          </div>
          <img src={`${cardImage[name]}`} alt={name} className='character-pic'/>
        </div>
        <div className='card-info'>
          <h4 className='character-name'>{name.toUpperCase()}</h4>
          <ul>
            { description }
          </ul>
        </div>
      </div>
    )
  }
}

Card.propTypes = {
  cardData: PropTypes.object.isRequired
}

export default Card;