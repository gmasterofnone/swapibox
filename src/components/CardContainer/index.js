import React from 'react';
import Card from '../Card'

const CardContainer = ({ cards }) => {
  const allCards = cards.map(card => <Card key={card.name} cardData={card}/>)
  return (
    <div className="card-container">
      { allCards }
    </div>
  )
}

export default CardContainer;