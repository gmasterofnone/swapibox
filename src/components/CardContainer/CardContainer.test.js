import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import CardContainer from '.';
import Card from '../Card'

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CardContainer data={[{name: 'Luke', type: 'people'},{name: 'Lando'},{name: 'R2-D2'}]} />);
  });


  it('matches the snapshot', () => { 
    expect(wrapper).toMatchSnapshot()
  });

  it('renders appropriate number of cards', () => {
    expect(wrapper.find(Card).length).toEqual(3)
  });

});