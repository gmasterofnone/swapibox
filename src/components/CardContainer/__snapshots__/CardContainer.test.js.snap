// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
      Array [
        Object {
          "name": "Luke",
        },
        Object {
          "name": "Lando",
        },
        Object {
          "name": "R2-D2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
            Object {
              "name": "Luke",
            }
          }
        />,
        <Card
          cardData={
            Object {
              "name": "Lando",
            }
          }
        />,
        <Card
          cardData={
            Object {
              "name": "R2-D2",
            }
          }
        />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke",
        "nodeType": "class",
        "props": Object {
          "cardData": Object {
            "name": "Luke",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Lando",
        "nodeType": "class",
        "props": Object {
          "cardData": Object {
            "name": "Lando",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "R2-D2",
        "nodeType": "class",
        "props": Object {
          "cardData": Object {
            "name": "R2-D2",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
              Object {
                "name": "Luke",
              }
            }
          />,
          <Card
            cardData={
              Object {
                "name": "Lando",
              }
            }
          />,
          <Card
            cardData={
              Object {
                "name": "R2-D2",
              }
            }
          />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke",
          "nodeType": "class",
          "props": Object {
            "cardData": Object {
              "name": "Luke",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Lando",
          "nodeType": "class",
          "props": Object {
            "cardData": Object {
              "name": "Lando",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "R2-D2",
          "nodeType": "class",
          "props": Object {
            "cardData": Object {
              "name": "R2-D2",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
